ui/components/component-library/button-base/button-base.scss

```diff
@@ -20,6 +20,13 @@
     height: 100%;
   }
 
+  &--ellipsis {
+    max-width: 100%;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
```

Comment:
Does this add more styles than the text component already has? Could we just change the root component to a `Text` component  instead of a `Box`?

Reply:
But couldn't you just change the root component to `<Text ellipses={elipses}` then just add the `max-width` styles?

![Screenshot 2023-03-16 at 4 53 58 PM](https://user-images.githubusercontent.com/8112138/225777577-9625e93c-24b8-4d4e-980c-f7a81f1e2f76.png)